### ISSUE TITLE
Support .editorconfig Glob Expressions

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#201]
To fully support `.editorconfig` files, not ony the parsing of an INI file (#199 ) is required, but also the support of [Glob Expressions](https://spec.editorconfig.org/index.html#glob-expressions).

| Special Characters | Matching|
|--------------------|----------------------
| `*`                | any string of characters, except path separators (/)
| `**`               | any string of characters
| `?`                | any single character, except path separators (/)
| `[seq]`            | any single character in seq
| `[!seq]`           | any single character not in seq
| `{s1,s2,s3}`       | any of the strings given (separated by commas, can be nested) (But {s1} only matches {s1} literally.)
| `{num1..num2}`     | any integer numbers between num1 and num2, where num1 and num2 can be either positive or negative

Without that, we can not determine which sections are relevant for specific files.

**Comment by sungam3r** [comment]
@Corniel Am I understand correctly that now I can’t configure the suppression of diagnostics through the editorconfig file? Like

```
[*.{props,csproj}]
dotnet_diagnostic.Proj0003.severity = suggestion
```

**Comment by Corniel** [comment]
> @Corniel Am I understand correctly that now I can’t configure the suppression of diagnostics through the editorconfig file? Like
> 
> ```
> [*.{props,csproj}]
> dotnet_diagnostic.Proj0003.severity = suggestion
> ```

Nope, you can't. That's because how MS Build works.  See [configuration](https://dotnet-project-file-analyzers.github.io/configuration.html) documentation.

**Comment by Frulfump** [comment]
> > [@Corniel](https://github.com/Corniel) Am I understand correctly that now I can’t configure the suppression of diagnostics through the editorconfig file? Like
> > ```
> > [*.{props,csproj}]
> > dotnet_diagnostic.Proj0003.severity = suggestion
> > ```
> 
> Nope, you can't. That's because how MS Build works. See [configuration](https://dotnet-project-file-analyzers.github.io/general/configuration.html) documentation.

The link gives me a 404

**Comment by Corniel** [comment]
I've updated the URL: https://dotnet-project-file-analyzers.github.io/configuration.html
